Mobile unit tests are failing on Gutenberg CI

The JS mobile unit tests are currently failing on the Gutenberg CI with the multiple instances of this error:

```
FAIL ../../packages/rich-text/src/component/test/index.native.js
  ● Test suite failed to run
    Cannot find module 'Platform' from 'react-native-implementation.js'

    However, Jest was able to find:
    	'../index.js'
    	'../index.native.js'

    You might want to include a file extension in your import, or update your 'moduleFileExtensions', which is currently ['js', 'json', 'jsx', 'ts', 'tsx', 'node'].

    See https://jestjs.io/docs/en/configuration#modulefileextensions-array-string

      4 | import { StyleSheet, TouchableOpacity, Text, View, Platform } from 'react-native';
      5 | 
    > 6 | const isAndroid = Platform.OS === 'android';
        |                   ^
      7 | const marginBottom = isAndroid ? -0.5 : 0;
      8 | const marginLeft = -3;
      9 | 
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:229:17)
      at Object.require [as Platform] (node_modules/react-native/Libraries/react-native/react-native-implementation.js:325:12)
      at Object.Platform (packages/components/src/button/index.native.js:6:19)
```

Example: [Travis Job failing](https://travis-ci.com/WordPress/gutenberg/jobs/230992153), on [this PR](https://github.com/WordPress/gutenberg/pull/17309)

The Travis job is set to "This job is allowed to fail." so it doesn't block the PRs but, we need to fix this in order to move to the next steps.

--- a/test/native/jest.config.js
+++ b/test/native/jest.config.js
@@ -38,6 +38,7 @@ module.exports = {
 		'/__device-tests__/',
 	],
 	testURL: 'http://localhost/',
+	// Add the `Libraries/Utilities` subfolder to the module directories, otherwise haste/jest doesn't find Platform.js on Travis
 	moduleDirectories: [ 'node_modules', 'node_modules/react-native/Libraries/Utilities' ],
 	moduleNameMapper: {
 		// Mock the CSS modules. See https://facebook.github.io/jest/docs/en/webpack.html#handling-static-assets

--- a/test/native/jest.config.js
+++ b/test/native/jest.config.js
@@ -38,7 +38,7 @@ module.exports = {
 		'/__device-tests__/',
 	],
 	testURL: 'http://localhost/',
-	moduleDirectories: [ 'node_modules' ],
+	moduleDirectories: [ 'node_modules', 'node_modules/react-native/Libraries/Utilities' ],
 	moduleNameMapper: {
 		// Mock the CSS modules. See https://facebook.github.io/jest/docs/en/webpack.html#handling-static-assets
 		'\\.(scss)$': '<rootDir>/' + configPath + '/__mocks__/styleMock.js',

